lldb/include/lldb/API/SBExpressionOptions.h

```diff
@@ -71,7 +70,7 @@ class LLDB_API SBExpressionOptions {
   /// Set the language using a pair of language code and version as
   /// defined by the DWARF 6 specification.
   /// WARNING: These codes may change until DWARF 6 is finalized.
-  void SetLanguage(lldb::SBSourceLanguageName name, uint32_t version);
+  void SetLanguage(lldb::SourceLanguageName name, uint32_t version);
```

Comment:
ABI break: Symbol changes names from `_ZN4lldb19SBExpressionOptions11SetLanguageENS_20SBSourceLanguageNameEj` to `_ZN4lldb19SBExpressionOptions11SetLanguageENS_18SourceLanguageNameEj`

Reply:
I replied but not to this comment.  If the meanings of the enum might change, the API is explicitly not stable.

Reply:
We can solve this by making making a typedef in SBDefine to keep the same symbol

Reply:
Changing the symbol name is an ABI break, changing the integral values would be an API break. My comment is only concerned about the former here.

Reply:
Without weighing in on the patch itself, I want to say that this is not how I would understand this distinction. Although a change in the value of a constant would not result in symbol changes in lldb, the value can get embedded in the code of the lldb user, and even (e.g. through its use as a template arg) in the names of their symbols.

In terms of compatibility, I think there are two levels that we can talk about:
- whether existing source code will work when it is compiled against new version of lldb (compile time)
- whether existing compiled code will work when used with new lldb (at runtime)

Changing a constant value breaks the second use case (just like a symbol name change would), although it would be harder to detect (which, in a way, makes it worse).

Reply:
That's true. I'm fairly certain our "ABI stability" guarantee applies to both scenarios, though I'm not sure how we can promise an "unstable API" while maintaining our "ABI stability" guarantee. I want to review as many SBAPI changes as possible to look for these kinds of ABI breaks but it doesn't scale very well.

Reply:
We don't have a formal policy about "provisional additions to the SB API".  The warning about the values change clearly means: "If you use this API, you will have to recompile against the version of lldb you intend to use."  So it is explicitly saying it's an unstable API.

Best would be to not do this, but if we find ourselves in a position where we have to, then we probably should require you to define `ALLOW_LLDB_UNSTABLE_SB_APIS` or some such, and guard unstable API's based on that.